### PR TITLE
Add additional checks for only remove the transaction id from source obj if not present

### PR DESCRIPTION
**Pull request by osulzhenko**

### CODE REVIEW

**Review by marki1an — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy [hunk 1]
```diff
@@ -646,6 +694,9 @@ class GppTransmitTidActivitiesSpec extends PrivacyBaseSpec {
             !bidderRequest.imp[0].ext.tid
             !bidderRequest.source.tid
         }
+
+        and: "Don’t affect source.schain if it was present"
+        bidderRequest.source.schain == sourceWithSchainAndTid.schain
```
[the remark this answers is not in the document]
Comment (osulzhenko) [reply src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy]: done, thanks

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy [hunk 2]
```diff
@@ -604,15 +646,21 @@ class GppTransmitTidActivitiesSpec extends PrivacyBaseSpec {
             !bidderRequest.imp[0].ext.tid
             !bidderRequest.source.tid
         }
+
+        and: "Don’t affect source.schain if it was present"
+        assert bidderRequest.source.schain == sourceWithSchainAndTid.schain
     }
 
-    def "PBS amp should remove bidRequest.(source/imp[0].ext).tid when ext.prebid.createTids=null and defaultAction=false"() {
+    def "PBS amp should remove bidRequest.(source/imp[0].ext).tid but don't change schain in request when ext.prebid.createTids=#createTid and defaultAction=false"() {
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy:654]: `#createTid` still static, meaning `null`
`PBS amp should remove bidRequest.(source/imp[0].ext).tid and don't change schain in request when ext.prebid.createTids=#createTid and defaultAction=false and schain specified in request`

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy [hunk 3]
```diff
@@ -165,17 +174,23 @@ class GppTransmitTidActivitiesSpec extends PrivacyBaseSpec {
...
+        and: "Don’t affect source.schain if it was present"
+        assert bidderRequest.source.schain == sourceWithSchainAndTid.schain
+
         where:
         createTid << [true, null]
     }
 
-    def "PBS auction should remove bidRequest.(source/imp[0].ext).tid when ext.prebid.createTids=null and activity disallowed"() {
-        given: "Bid requests with TID fields and account id"
+    def "PBS auction should remove bidRequest.(source/imp[0].ext).tid but don't change schain in request when ext.prebid.createTids=null and activity disallowed"() {
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy:184]: `PBS auction should remove bidRequest.(source/imp[0].ext).tid and don't change schain in request when ext.prebid.createTids=null and activity disallowed`

Please check other places.

**Review by marki1an — APPROVED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy (hunk 3 above)
Comment (osulzhenko) [reply src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy]: updated

File: src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy (hunk 2 above)
Comment (osulzhenko) [reply src/test/groovy/org/prebid/server/functional/tests/privacy/GppTransmitTidActivitiesSpec.groovy]: updated

**Review by marki1an — APPROVED** [review]

**Pull request merged**